docs/reference/ingest/processors/enrich.asciidoc

```diff
@@ -0,0 +1,387 @@
+[role="xpack"]
+[testenv="basic"]
+[[enrich-processor]]
+=== Enrich Processor
+
+The enrich processor allows documents to be enriched with data from an enrich index
+that is managed by an enrich policy prior to indexing.
+
+The data that is used by the enrich index is managed by the user in a regular indices.
+An enrich policy is configuration that indicates how an enrich index is created from
+the data  in the user's maintained indices. When an enrich policy is executed
+a new enrich index is created for that policy, which the enrich process can then use.
+
+An enrich policy also controls what kind of enrichment the `enrich` processor is able to do.
+
+[[enrich-options]]
+.Enrich Options
+[options="header"]
+|======
+| Name               | Required  | Default              | Description
+| `policy_name`      | yes       | -                    | The name of the enrich policy to use.
+| `enrich_key`       | no        | Policy enrich_key    | The field to get the value from for the enrich lookup.
+| `ignore_missing`   | no        | `false`              | If `true` and `enrich_key` does not exist, the processor quietly exits without modifying the document
+| `enrich_values`    | yes       | -                    | Describes what fields should be added to the document being indexed from the lookup document
+include::common-options.asciidoc[]
+|======
+
+[[enrich-policy-definition]]
+==== Enrich Policy Definition
+
+The enrich processor requires more than just the configuration in a pipeline.
+The main piece to configure is the enrich policy:
+
+[[enrich-policy-options]]
```

Comment:
this feels that this should live outside of the processor doc. 
Perhaps a top level item "Enriching data" in Ingest heading https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html ?

Reply:
++ to this, like how `Conditional Execution in Pipelines` has its own call out / section, right @jakelandis 	?

Reply:
Do you mean only the `Enrich Policy Definition` section or also the getting started and description of APIs?

Reply:
yes, very much like `Conditional Execution in Pipelines` where there can multiple sub pages to describe the Enriching process. The processor definition would still exist, but would be a pretty boring reference. 

This PR doesn't need to include the entirety of the doc, but I think having a separate top level where one sub item can be the policy definition doc would be a nice way to present this data.

Reply:
so to actually answer your question... the processor reference doc would go with the processors, the getting started, policy, and API definitions would all be sub headers under `Enriching ingest data` (or something like that).

Reply:
I've pushed: https://github.com/elastic/elasticsearch/pull/45084/commits/9ed37ece49e39d3d60cc5c7689d1c9942a3c0efe